helper/dbtxn/dbtxn.go

```diff
@@ -0,0 +1,111 @@
+package dbtxn
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+type Config struct {
+	Name       string
+	Username   string
+	Password   string
+	Expiration string
+}
+
+// ExecuteDBQuery handles executing one single statement, while properly releasing its resources.
+// - ctx: 	Optional, may be nil
+// - db: 	Required
+// - config: 	Optional, may be nil
+// - query: 	Required
+func ExecuteDBQuery(ctx context.Context, db *sql.DB, config *Config, query string) error {
+
+	parsedQuery := parseQuery(config, query)
+
+	stmt, err := dbPrepare(ctx, db, parsedQuery)
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	return execute(ctx, stmt)
+}
+
+// ExecuteTxQuery handles executing one single statement, while properly releasing its resources.
+// - ctx: 	Optional, may be nil
+// - tx: 	Required
+// - config: 	Optional, may be nil
+// - query: 	Required
+func ExecuteTxQuery(ctx context.Context, tx *sql.Tx, config *Config, query string) error {
+
+	parsedQuery := parseQuery(config, query)
+
+	stmt, err := txPrepare(ctx, tx, parsedQuery)
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	return execute(ctx, stmt)
+}
+
+func dbPrepare(ctx context.Context, db *sql.DB, parsedQuery string) (*sql.Stmt, error) {
```

Comment:
This is only used in one place, can you just move the logic up? Same with txPrepare.

Reply:
Sure!